tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/lambda/FunctionPointersNegativeTest.java

```diff
@@ -34,9 +34,11 @@
     public void testFunctionPointerAsVariable() {
         CompileResult result =
                 BCompileUtil.compile("test-src/expressions/lambda/negative/fp-type-mismatch1-negative.bal");
-        Assert.assertEquals(result.getErrorCount(), 1);
+        Assert.assertEquals(result.getErrorCount(), 3);
         BAssertUtil.validateError(result, 0, "incompatible types: expected 'function (string,int) returns " +
                 "(boolean)', found 'function (string,float) returns (boolean)'", 2, 53);
+        BAssertUtil.validateError(result, 1, "unknown type 'Context'", 10, 29);
+        BAssertUtil.validateError(result, 2, "unknown type 'FunctionEntry'", 12, 5);
```

Comment:
Is it correct that this error is getting logged? Only `Context` is an unknown type here, right? Even though there are errors when attempting to resolve `FunctionEntry`'s members, it is not unknown?

Reply:
This error message at index 2 is due to the behavior of tuple, the behavior added by this PR for function type is similar.

Sample for SLP preview 3:
```

type A object {
    K i;

    function __init(K i) {
        self.i = i;
    }
};

type B [A, K];

type C B|A|D;

type D record {
    K i;
    B j;
    C k;
};
``` 
output:
```
error: .::main.bal:10:9: unknown type 'A'
error: .::main.bal:10:12: unknown type 'K'
error: .::main.bal:12:10: unknown type 'A'
error: .::main.bal:12:12: unknown type 'D'
error: .::main.bal:3:5: unknown type 'K'
error: .::main.bal:15:5: unknown type 'K'
error: .::main.bal:16:5: unknown type 'B'
error: .::main.bal:17:5: unknown type 'C'
error: .::main.bal:5:21: unknown type 'K'
```

Reply:
The output doesn't seem consistent right? An "unknown type 'B'" error is only logged for the usage as a field in `D` and not for the definition of `C`?

For the following sample, there is only one error.

```ballerina
type FunctionType record {
    Context c;
};

type FunctionEntry [FunctionType, typedesc<anydata>];
map<FunctionEntry> functions = { };
```

```cmd
error: .::test.bal:2:5: unknown type 'Context'
```

I guess we can go ahead with these changes for the time-being. But shall we create an issue to revisit this and make the errors consistent?

Reply:
Issue created: https://github.com/ballerina-platform/ballerina-lang/issues/25831